frontend/lib/Views/AdminPage/groups_view.dart

```diff
@@ -64,34 +79,32 @@ class _GroupsViewState extends State<GroupsView> {
         ),
       );
 
-  Widget _buildBody() => RefreshIndicator(
-        onRefresh: () => widget.gateway.getGroupsSummary().onError(onError),
-        child: FutureBuilder<List<GroupSummary>>(
-          future: widget.gateway.getGroupsSummary().onError(onError),
-          builder: (context, snapshot) {
-            if (snapshot.connectionState == ConnectionState.none ||
-                snapshot.data == null) {
-              return loadingInCenter();
-            }
-            return ChangeNotifierProvider(
-              create: (context) =>
-                  GroupListProvider(snapshot.data!, widget.gateway),
+  FutureBuilder _buildBody() => FutureBuilder(
+        future: widget.gateway.getGroupsSummary(),
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.none ||
+              snapshot.data == null) {
+            return loadingInCenter();
+          } else {
+            groupListProvider.makeGroupList(snapshot.data!);
```

Comment:
Zastanawiam się, czy skoro sprawdzamy wcześniej, czy `snapshot.data` jest `null'em`, to czy musimy używać `!`? :thinking: